src/index.ts

```diff
@@ -1,4 +1,137 @@
-export default () => {
-  // eslint-disable-next-line no-console
-  console.log('https://www.youtube.com/watch?v=dQw4w9WgXcQ')
+import ResponseError from './ResponseError'
+
+export interface ApiClientConfiguration {
+  baseUrl: string
+  headers: Record<string, string>
 }
+
+export interface RequestOptions {
+  baseUrl?: string
+  headers?: Record<string, string>
+  params?: Record<string, string | number>
+  accessToken?: string
+}
+
+class ApiClient {
+  private configuration: ApiClientConfiguration = {
+    baseUrl: '',
+    headers: { 'Content-Type': 'application/json' },
+  }
+
+  private static replaceParameters(path: string, options?: RequestOptions) {
+    const parameters = path.match(/{(.*?)}/g) || []
+
+    if (!parameters.length) {
+      return path
+    }
+
+    let replacedPath = path
+    parameters.forEach((param) => {
+      const value = options?.params?.[param.slice(1, -1)]
```

Comment:
I think you should have the param directly in capture group 1 if present (there is a way to name them iirc)

Reply:
only works with non global regular expressions and the browser support for `matchAll` is very weak